src/zenml/stack/utils.py

```diff
@@ -139,3 +140,33 @@ def get_flavor_by_name_and_type_from_zen_store(
             f"'{component_type}' exists."
         )
     return flavors[0]
+
+
+class stack_context:
```

Comment:
We have actually implemented [a very similar context manager](https://github.com/zenml-io/zenml/blob/fc6f665bcab3a9882ac577cda5bddf6f5dded9cd/src/zenml/cli/utils.py#L2484) to this one already. In contrast to this implementation, it takes in the target stack id or name as an input. IMO, this makes it a bit clearer implementation-wise, as the context manager takes on the responsibility to switch the stack back and forth and you do not have to deal with this switch somewhere else in the code. I would recommend moving this implementation out of the `cli.utils` to our main `utils` module and adapting the pipeline code accordingly to use this implementation.